### PR TITLE
Add Tool Interaction Views

### PR DESCRIPTION
**Pull request by LeonNissen**
# Add Tool Interaction Views

## :gear: Release Notes 
With this release we add two new roles `.assistantToolCall` and `.assistantToolResponse` which will be visible in the chat.
This will provides the user with additional context about the underlying techniques/functions used during the chat.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [ ] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Sources/SpeziChat/Models/ChatEntity.swift [hunk 1]
```diff
@@ -19,13 +19,17 @@ public struct ChatEntity: Codable, Equatable, Hashable, Identifiable {
     public enum Role: Codable, Equatable, Hashable {
         case user
         case assistant
+        case assistantToolCall
+        case assistantToolResponse
```
Comment (PSchmiedmayer) [inline Sources/SpeziChat/Models/ChatEntity.swift:23]: Thank you for the additions here.

We should create an extensive issue that documents a future extension into a more generalized setup where we can use static functions on the Role type that will be transformed into struct conforming to View to allow an easy extension or additional types for everyone that is using Spezi chat while we provide default implementations for the roles that we have detailed here.

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements; it does a non-breaking fix for what we want to do. In the long run, we should add a more extensible structure for custom views in the chat.

I would also defer to @philippzagar for some additional input.

File: Sources/SpeziChat/Models/ChatEntity.swift (hunk 1 above)
Comment (philippzagar) [reply Sources/SpeziChat/Models/ChatEntity.swift:23]: Fully agree with Paul here, we should do that in an extensive GitHub issue.

File: Sources/SpeziChat/ChatView+SpeechButton.swift [hunk 2]
```diff
@@ -86,14 +86,14 @@ extension View {
 
 #if DEBUG
 #Preview {
-    @State var chat: Chat = .init(
+    @Previewable @State var chat: Chat = .init(
```
Comment (philippzagar) [inline Sources/SpeziChat/ChatView+SpeechButton.swift:89]: Quickly made these changes in order to resolve warnings!

File: Sources/SpeziChat/Resources/Localizable.xcstrings [hunk 3]
```diff
@@ -20,6 +20,9 @@
           }
         }
       }
+    },
+    "Equal sign" : {
```
Comment (philippzagar) [inline Sources/SpeziChat/Resources/Localizable.xcstrings:24]: Missing translations?

File: Sources/SpeziChat/ToolInteractionView.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+/// The view that is represented when a tool call or tool reponse based on the ``ChatEntity/Role``.
+struct ToolInteractionView: View {
```
Comment (philippzagar) [inline Sources/SpeziChat/ToolInteractionView.swift:12]: We should add a small UI test for the new tool interactions view

**Review by philippzagar — CHANGES_REQUESTED** [review]
Thanks for the PR @LeonNissen! 🚀 
As @PSchmiedmayer already mentioned: We should definitely create a GitHub issue that lays out the future extension of `Role`s into a more generalized setup.
Also, please add a small UI test to keep up the coverage tests that are failing

**Review by philippzagar — APPROVED** [review]
Thanks @LeonNissen for this PR and adding the tests, looks graet! 🚀

**Comment by philippzagar** [comment]
@LeonNissen Just noticed that the commits on this branch are not properly signed, therefore preventing a merge. 
Two options: Either you retroactively sign the commits and force push to the respective branch, or @PSchmiedmayer quickly merges the PR by bypassing the merge checks.

**Pull request merged**